docs/source/developers/guide/about_arrow.rst

```diff
@@ -0,0 +1,37 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+
+.. SCOPE OF THIS SECTION
+.. This section should include basic information
+.. about the Arrow project. If possible the content should be
+.. written in words easy understandable to beginner notß
+.. necessary acquainted with the library and the technical
+.. details.
+
+.. _about_arrow:
+
+***********
+About Arrow
+***********
+
+General overview. TODO
```

Comment:
I think an "About Arrow" page (if we need one) would not live in the developers section but rather at the top level.

Reply:
That said, if https://arrow.apache.org/ and https://arrow.apache.org/overview/ are not informative enough, we should perhaps try to improve them instead of adding another resource here.

Reply:
I will make a separate ticket to improve **About Arrow** and link this part in the guide. I will also change the title to **Architectural Overview** as this is the actual scope of this section.